tests/moduletool/test_repos.py

```diff
@@ -43,6 +46,26 @@ def test_remote_repo_good(git_modules_dir, modules_repo):
     result = repo.clone("test-repository", coroot)
     assert result
     assert os.path.exists(os.path.join(coroot, "test-repository", "README"))
+    shutil.rmtree(os.path.join(coroot, "test-repository"))
+
+
+def test_remote_repo_good2(git_modules_dir, modules_repo):
+    repo = RemoteRepo("https://github.com/rmccue/{}")
+    coroot = os.path.join(git_modules_dir, "clone_remote_good")
+    result = repo.clone("test-repository", coroot)
+    assert result
+    assert os.path.exists(os.path.join(coroot, "test-repository", "README"))
+    shutil.rmtree(os.path.join(coroot, "test-repository"))
```

Comment:
It's not a good idea to write content into the directory provided by the `git_modules_dir` fixture. This fixture is a session-scoped fixture. This means that the changes made by you test case will be passed onto the next test case. Adding `shutil.rmtree()` at the end of the test case is not sufficient, as this line will be skipped when the test case raises an exception. This way you get the situation where one failing test case causes several other to fail as well. This makes the debugging of a failing test case a lot more difficult.

In the above-mentioned test cases you only need a temporary directory to perform to clone a Git repository into. In that case it's better to use the `tmpdir` fixture. This is a function scoped fixture that provides an empty temporary directory. This fixture will automatically remove all the content in the tmpdir at the end of each test case. So there is not need to do an `shutil.rmtree()` at the end of the test case.